### PR TITLE
add task 6

### PR DESCRIPTION
**Pull request by suzik86**

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -9,6 +9,8 @@ import { AppComponent } from './app.component';
 import { CoursesPageModule } from './pages/courses-page/courses-page.module';
 import { LoginPageModule } from './pages/login-page/login-page.module';
 import { SharedModule } from './shared/shared.module';
+import { AddCoursePageModule } from './pages/add-course-page/add-course-page.module';
+
```
Comment (katrin2404) [inline src/app/app.module.ts]: Remove extra lines.

File: src/app/pages/add-course-page/add-course-page.component.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+export class AddCoursePageComponent {
+  title = '';
+  creationDate = '';
+  duration = '';
+  description = '';
+
+  constructor(private coursesService: CoursesService) {}
+
+  onSave(): void {
+    console.log(this.title, 'was created!');
```
Comment (katrin2404) [inline src/app/pages/add-course-page/add-course-page.component.ts:18]: Could you please display courses list after successful saving course?

File: src/app/pages/add-course-page/add-course-page.module.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    AddCoursePageComponent,
+    DateComponent,
+    DurationComponent,
+  ],
+  exports: [
+    CommonModule,
+    AddCoursePageComponent,
+    MatButtonModule,
+  ],
+  providers: [
```
Comment (katrin2404) [inline src/app/pages/add-course-page/add-course-page.module.ts]: Remove unused providers property.

File: src/app/pages/add-course-page/add-course-page.module.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { MatButtonModule } from '@angular/material/button';
+import { MatInputModule } from '@angular/material/input';
+import { SharedModule } from 'src/app/shared/shared.module';
+import { AddCoursePageComponent } from './add-course-page.component';
+import { DateComponent } from './components/date/date.component';
+import { DurationComponent } from './components/duration/duration.component';
+
```
Comment (katrin2404) [inline src/app/pages/add-course-page/add-course-page.module.ts:9]: Leave just one empty row.

File: src/app/pages/add-course-page/components/date/date.component.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'app-date',
+  templateUrl: './date.component.html',
+  styleUrls: ['./date.component.scss'],
+})
+export class DateComponent {
+  @Input() creationDate: string;
+
```
Comment (katrin2404) [inline src/app/pages/add-course-page/components/date/date.component.ts]: Remove empty line

File: src/app/pages/courses-page/components/course/course.component.ts [hunk 6]
```diff
@@ -7,6 +7,7 @@ import { DeleteCourseModalComponent } from './delete-course-modal/delete-course-
   selector: 'app-course',
   templateUrl: './course.component.html',
   styleUrls: ['./course.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class CourseComponent implements OnInit{
```
Comment (katrin2404) [inline src/app/pages/courses-page/components/course/course.component.ts:12]: Opening of deleting modal should be in smart component. You need to use event binding here.

File: src/app/pages/add-course-page/add-course-page.module.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    MatButtonModule,
+    MatInputModule,
+  ],
+  declarations: [
+    AddCoursePageComponent,
+    DateComponent,
+    DurationComponent,
+  ],
+  exports: [
+    CommonModule,
```
Comment (katrin2404) [inline src/app/pages/add-course-page/add-course-page.module.ts]: You can export just AddCoursePageComponent.

File: src/app/app.module.ts (hunk 1 above)
Comment (suzik86) [reply src/app/app.module.ts]: done

File: src/app/pages/add-course-page/add-course-page.module.ts (hunk 4 above)
Comment (suzik86) [reply src/app/pages/add-course-page/add-course-page.module.ts:9]: done

File: src/app/pages/add-course-page/components/date/date.component.ts (hunk 5 above)
Comment (suzik86) [reply src/app/pages/add-course-page/components/date/date.component.ts]: done

File: src/app/pages/add-course-page/add-course-page.module.ts (hunk 7 above)
Comment (suzik86) [reply src/app/pages/add-course-page/add-course-page.module.ts]: done

File: src/app/pages/add-course-page/add-course-page.module.ts (hunk 3 above)
Comment (suzik86) [reply src/app/pages/add-course-page/add-course-page.module.ts]: done

**Pull request merged**